src/utils.rs

```diff
@@ -0,0 +1,34 @@
+use macroquad::{
+    experimental::collections::storage,
+    prelude::{get_frame_time, get_internal_gl, InternalGlContext, Vec2},
+    rand::gen_range,
+};
+use macroquad_particles::Emitter;
+
+use crate::Resources;
+use std::f32;
+
+pub fn explode(position: Vec2, radius: f32) {
+    //Center position, radius of explosion. Only FX at the moment
+    let mut r = 0.0;
+
+    let mut resources = storage::get_mut::<Resources>();
+    while r < radius - 5. {
+        // Particles are 5 in radius
+        r += 10.0 / (r + 1.);
+        let angle = gen_range(0.0, f32::consts::PI * 2.);
+        resources
+            .fx_cannonball_hit
```

Comment:
name `fx_cannonball` assumes that the fx is used only by the cannon weapon
while `explode` function looks like it is designed for use in any weapon

Maybe `explode` should be moved to a cannon weapon?

Reply:
I've just kept the name out of convention, since that's what it was named previously :) Nothing else uses it now so renaming it should be no issue